File: lib/domain/entities.ts

export interface UserProfile {
  name: string;
  identity: string;
  email: string;
  phone: string;
  dob?: string;
  firebaseId?: string;
}

export interface RemoteConfig {
  syncThreshold: number;
  enablePremiumFeatures: boolean;
  activeCampaign: string;
}

export interface ProductivityEvent {
  name: "Hola_mundo";
  properties: {
    years_mobile_experience: number;
    years_flutter_experience: number;
    published_apps: number;
  };
}

export enum BlocStatus {
  INITIAL = 'INITIAL',
  LOADING = 'LOADING',
  SUCCESS = 'SUCCESS',
  ERROR = 'ERROR'
}

export type AppView = 'DASHBOARD' | 'CONFIG' | 'LOGS';

export interface GrowthState {
  status: BlocStatus;
  currentView: AppView;
  user?: UserProfile;
  tasks: string[];
  logs: string[];
  config: RemoteConfig;
  permissions: {
    camera: 'granted' | 'denied' | 'prompt';
    location: 'granted' | 'denied' | 'prompt';
  };
  errorMessage?: string;
}

File: lib/domain/entities.dart

import 'package:equatable/equatable.dart';

class UserProfile extends Equatable {
  final String name;
  final String identity;
  final String email;
  final String phone;
  final String? dob;
  final String? firebaseId;

  const UserProfile({
    required this.name,
    required this.identity,
    required this.email,
    required this.phone,
    this.dob,
    this.firebaseId,
  });

  UserProfile copyWith({String? dob, String? firebaseId}) {
    return UserProfile(
      name: name,
      identity: identity,
      email: email,
      phone: phone,
      dob: dob ?? this.dob,
      firebaseId: firebaseId ?? this.firebaseId,
    );
  }

  @override
  List<Object?> get props => [name, identity, email, phone, dob, firebaseId];
}

class AppConfig extends Equatable {
  final bool enablePremium;
  final String activeCampaign;

  const AppConfig({required this.enablePremium, required this.activeCampaign});

  @override
  List<Object?> get props => [enablePremium, activeCampaign];
}
